README.md

```diff
@@ -39,15 +65,35 @@ You need the `-n` to stop hoogle from trying to use https locally. You will need
 
 ### Add external tool to `default.nix`
 
-The sources you pull down with `niv` are accessible under `sources.<name>`. Here is an example adding the specific version of `ghcid` we fetched earlier to our development shell:
+The sources you pull down with `niv` are accessible under `sources.<name>`. Its a derivation with the attributes you need to fetch the source, not the source itself. If you want to pull in a source you need to either give it to a function that knows how to fetch the contents like `callCabal2nix` or if the source has a `default.nix` you can import it directly, like so: `import sources.niv {}`.
+
+---
+
+Often you will want to explore what you have just imported since you may only want one of its attributes, you can do this by adding the source as an exported attribute in your `default.nix`:
+
+```
+in
+  if pkgs.lib.inNixShell then shell else {
+    inherit exe;
+    inherit docker;
+    ### Added here
+    src = import sources.niv {};
+  }
+
+```
+
+Then call `nix repl default.nix` and you can tab complete `src.<tab>` to see what attributes are inside. (Note, this is how I know to call `(import sources.niv {}).niv` to get the `niv` derivation).
+
```

Comment:
This section felt a bit hacky. We should suggest them to `nix repl`, but saying "you can modify the source code temporarily to try this" feels a bit odd.

Reply:
I agree telling users to modify the source isn't as nice. However, I don't see a way around this one. I'm actually using the `niv` executable as an example to show how you would debug an imported source. Users would need to add `src = import sources.<name> {};` where `name` is whatever they want to debug is called in `sources.json`. In other words, we cannot write it for them because we don't know what they are going to import.

I can adjust it to be `<name>` instead of `niv` to make it more clear what you actually need to do.